Python/gc_free_threading.c

```diff
@@ -1160,8 +1162,11 @@ gc_collect_main(PyThreadState *tstate, int generation, _PyGC_Reason reason)
             n+m, n, d);
     }
 
-    // Clear free lists in all threads
-    _PyGC_ClearAllFreeLists(interp);
+    // Clear the current thread's free-list again.
+    HEAD_LOCK(&_PyRuntime);
```

Comment:
This doesn't need a `HEAD_LOCK()` because we aren't traversing the linked list of threads.

Reply:
Ah you are right :)